ZUGFeRD/TradeAllowanceCharge.cs

```diff
@@ -56,6 +56,11 @@ public class TradeAllowanceCharge : Charge
         /// </summary>
         public string Reason { get; set; }
 
+        /// <summary>
+        /// Code for the reason for allowance or charge
+        /// </summary>
+        public string ReasonCode { get; set; }

```

Comment:
It could be a code from UNTDID 7161 or UNTDID 5189 or PEPPOL subset.
According to business rule BR-CO-22 you can use the reason code or  the reason text or both:

"Geschäftsregel: BR-CO-22 Zuschläge auf Dokumentenebene
Jeder Zuschlag auf Dokumentenebene (BG-21) muss einen Grund für den Zuschlag auf Dokumentenebene (BT-
104) oder einen Code des Grundes für den Zuschlag auf Dokumentenebene (BT-105) oder beides enthalten."

According to the documentation the reason could be a free text. So I don't know, if the reason text could only one of the predefined text of any code. Otherwise the text can be in any language e.g. in German.

For me it's only important to read the reason code from an incoming invoice (I don't need to create an invoice).
So I could change the setter to an internal setter.